tests/test_doublyrobust.py

```diff
@@ -315,6 +335,266 @@ def test_sklearn_in_tmle_missing(self, mf):
+        with pytest.raises(ValueError):
+            stmle.fit(p=[0.1], conditional=["df['male']==1", "df['male']==0"])
+
+        with pytest.raises(ValueError):
+            stmle.fit(p=[0.1, 0.3], conditional=["df['male']==1"])
+
+    def test_error_summary(self, df):
+        stmle = StochasticTMLE(df=df, exposure='art', outcome='dead')
+        stmle.exposure_model('male + age0 + age_rs1 + age_rs2 + cd40 + cd4_rs1 + cd4_rs2 + dvl0')
+        stmle.outcome_model('male + age0 + age_rs1 + age_rs2 + cd40 + cd4_rs1 + cd4_rs2 + dvl0')
+        with pytest.raises(ValueError):
+            stmle.summary()
+
+    def test_warn_missing_data(self):
+        df = pd.DataFrame()
+        df['A'] = [1, 1, 0, 0, np.nan]
+        df['Y'] = [np.nan, 0, 1, 0, 1]
+        with pytest.warns(UserWarning):
+            StochasticTMLE(df=df, exposure='A', outcome='Y')
+
+    def test_drop_missing_data(self):
+        df = pd.DataFrame()
+        df['A'] = [1, 1, 0, 0, np.nan]
+        df['Y'] = [np.nan, 0, 1, 0, 1]
+        stmle = StochasticTMLE(df=df, exposure='A', outcome='Y')
+        assert stmle.df.shape[0] == 3
+
+    def test_continuous_processing(self):
+        a_list = [0, 1, 1, 0, 1, 1, 0, 0]
+        y_list = [1, -1, 5, 0, 0, 0, 10, -5]
+        df = pd.DataFrame()
+        df['A'] = a_list
+        df['Y'] = y_list
+
+        stmle = StochasticTMLE(df=df, exposure='A', outcome='Y', continuous_bound=0.0001)
+
+        # Checking all flagged parts are correct
+        assert stmle._continuous_outcome is True
+        assert stmle._continuous_min == -5
+        assert stmle._continuous_max == 10
+        assert stmle._cb == 0.0001
+
+        # Checking that TMLE bounding works as intended
+        y_bound = [2 / 5, 4 / 15, 2 / 3, 1 / 3, 1 / 3, 1 / 3, 0.9999, 0.0001]
+        pdt.assert_series_equal(pd.Series(y_bound),
+                                stmle.df['Y'],
+                                check_dtype=False, check_names=False)
+
+    def test_marginal_vector_length_stoch(self, df):
+        stmle = StochasticTMLE(df=df, exposure='art', outcome='dead')
+        stmle.exposure_model('male')
+        stmle.outcome_model('art + male + age0')
+        stmle.fit(p=0.4, samples=7)
+        assert len(stmle.marginals_vector) == 7
+
+    def test_qmodel_params(self, simple_df):
+        # Comparing to SAS logit model
+        sas_params = [-1.0699, -0.9525, 1.5462]
+        sas_preds = [0.3831332, 0.2554221, 0.1168668, 0.2554221, 0.6168668, 0.6168668, 0.1168668, 0.2554221, 0.3831332]
+
+        stmle = StochasticTMLE(df=simple_df, exposure='A', outcome='Y')
+        stmle.outcome_model('A + W')
+        est_params = stmle._outcome_model.params
+        est_preds = stmle._Qinit_
+
+        npt.assert_allclose(sas_params, est_params, atol=1e-4)
+        npt.assert_allclose(sas_preds, est_preds, atol=1e-6)
+
+    def test_qmodel_params2(self, simple_df):
+        # Comparing to SAS linear model
```

Comment:
if possible, I suggest putting a reproducible SAS command here. It'll save you time later when/if you have to reproduce it.